### PR TITLE
raspberry magic number begone

### PR DESCRIPTION
**Pull request by perigoso**
a second round of cleanup on adi code, this fixes the misleading idcode naming, we were actually reading the DP DPIDR register, it also fixes the JEP106 code for raspberry, and the handling of designer code in a couple registers (the reason for the wrong jep code)

### CODE REVIEW

**Comment by dragonmux** [comment]
We'll let you figure out why the build broke before we review this, and we'll make sure to test against our RP2040 to prevent any possible regressions we might expect

**Comment by perigoso** [comment]
should be good, added a couple commits that changed/removed some superfluous checks

I also want to rename t->idcode to maybe t->partno, as it actually inherits the ap->partno, nothing to do with idcode, but I'll leave that for another day

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:648]: This `#else` branch should be eliminated.

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:652]: this `else` should be in the `#if` block. The {}'s are valid regardless

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:603]: This `#else` block should be eliminated

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:607]: This `else` should be in the above `#if` block

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:332]: The braces around the body of this if block should be eliminated

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:135]: Suggest putting this in a `/* clang-format: off */` block and formatting it so the indentation is in the source but not inside the strings. This will save us some space int the firmware and make it easier to read

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:74]: `U` on these constants please

**Review by dragonmux — CHANGES_REQUESTED** [review]
This is already looking really good - unfortunate that in reformatting `cortexm.c`, it makes our review look really harsh, but most of the points raised should be simple fixes.

We're happy with how this does what it does and will test and merge once the major review points are addressed.

**Comment by dragonmux** [comment]
Going to suggest that renaming idcode -> target_id should be a separate PR based on this one. We'd like to get this merged and see how it plays out with the other big 6, keeping concerns separated as best as possible (it's a good change though and lends some much needed clarity to the code)

File: src/target/adiv5.h
Comment (perigoso) [reply src/target/adiv5.h:222]: DPIDR is actually part of the adi spec, `DPIDR, Debug Port Identification Register`, but i agree it's not immediately clear

File: src/target/cortexm.c
Comment (perigoso) [reply src/target/cortexm.c:1669]: uh, yeah that doesnt feel right

File: src/target/adiv5.h
Comment (dragonmux) [reply src/target/adiv5.h:222]: Ah, sorry, should have made ourself clearer.. it's not in the ARMv7-M TRM or the Cortex-M4 TRM (for example), so you'd have to be looking at just the right spec to find the term, which is probably not the spec you're reading if you're trying to find out how values are encoded and at what addresses they can be found (that's in the ARMv7 TRM debugging section, and core-specific TRMs)

Comment (perigoso) [reply src/target/adiv5.h:222]: right, I was looking at the ADIv5 and ADIv6 specs, debu_port_id sounds good enough, as long as its doesnt imply something like the idcode did

File: src/target/cortexm.c
Comment (dragonmux) [reply src/target/cortexm.c:1669]: The other instance of clang-format doing things we don't want was in `command.c` where it tried to put the command array initialiser closing bracket on the same line as the sentinel, and then also put the first item of the array on the same line as the array's defined and opened.. need to figure out why and fix it

Comment (perigoso) [reply src/target/cortexm.c]: I manually reverted back this format, but future formats may try to do it again

File: src/target/cortexm.c
Comment (perigoso) [reply src/target/cortexm.c]: I added an extra commit moving these, it made sense

File: src/target/cortexm.c
Comment (perigoso) [reply src/target/cortexm.c]: renamed them to SEMIHOSTING_SYS_etc to avoid conflicts

File: src/target/cortexm.c
Comment (perigoso) [reply src/target/cortexm.c]: fyi `/* clang-format: off */` doesnt work, must be `/* clang-format off */` TIL too

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:1204]: The extra perens should be removed

**Review by dragonmux — CHANGES_REQUESTED** [review]
Spotted just one last thing that needs fixing but otherwise this LGTM and is ready for merge. We'll test this against our RPi Pico and approve for merge once the one item is fixed.

**Review by dragonmux — DISMISSED** [review]
LGTM, we'll merge this once the 6 PRs that are ahead of this are in and we've been able to properly test this

**Comment by perigoso** [comment]
Tested probe (hosted and native) with success on:
- Nordic nRF52
- SAMS70J19A
- Atmel SAM3U
- 1 Unknown ARM Cortex-M Designer 0x423 Part ID 0x150 (Renesas RA6M2)
- RP2040

**Comment by dragonmux** [comment]
Please rebase this then we'll merge it.

**Review by dragonmux — APPROVED** [review]
LGTM! Testing against Tiva-C then merging.

**Pull request merged**
